### PR TITLE
add apple app store engine

### PR DESCRIPTION
**Pull request by liimee**
## What does this PR do?

Adds the apple app store engine

## Why is this change important?

I just feel that this engine is missing

## How to test this PR locally?

search `!aps twitter`

## Related issues

___

sorry for submitting two PRs at the same time :sweat_smile:

### CODE REVIEW

**Comment by liimee** [comment]
They seem to provide price data for paid apps -- but I don't think it would work without setting the template to `products.html`

File: searx/engines/apple_app_store.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    json_result = loads(resp.text)
+
+    for result in json_result['results']:
+        results.append(
+            {
+                'url': result['trackViewUrl'],
+                'title': result['trackName'],
+                'content': result['description'],
+                'img_src': result['artworkUrl100'],
+                'thumbnail': result['screenshotUrls'][0],
```
Comment (dalf) [inline searx/engines/apple_app_store.py]: The other engines in the `app` category doesn't provide a screenshot : 
image

This make the "apple_app_store" engine huge compare to others.

So, can you remove the thumbnail?

**Review by dalf — CHANGES_REQUESTED** [review]
Except the thumbnail, LGTM to me, thank you!

File: searx/engines/apple_app_store.py (hunk 1 above)
Comment (liimee) [reply searx/engines/apple_app_store.py]: I thought it'd be nice if we could get a quick peek of the app before actually opening the URL, but ok

Comment (liimee) [reply searx/engines/apple_app_store.py]: done

Comment (dalf) [reply searx/engines/apple_app_store.py]: > I thought it'd be nice if we could get a quick peek of the app before actually opening the URL

I agree; the thing is, I don't know how to display these screenshots.
I thought about something similar to the image: when the users click on a result, the screenshot(s) are displayed on the right side, so the users can still browse the result quickly without too much information in the result list.

Of course it requires more development (the embedded videos could be displayed on the right side too).

**Pull request merged**